internal/store/util.go

```diff
@@ -32,3 +40,32 @@ func sliceToMap(inc []string) map[string]struct{} {
 
 	return m
 }
+
+func filterMap[V any](m map[string]V, keys []string, mode filterMode) map[string]V {
+	result := maps.Clone(m)
+
+	keySet := make(map[string]struct{}, len(keys))
+	for _, k := range keys {
+		keySet[strings.ToLower(k)] = struct{}{}
+	}
+
+	switch mode {
+	case include:
+		maps.DeleteFunc(result, func(k string, _ V) bool {
+			_, exists := keySet[strings.ToLower(k)]
+			return !exists
+		})
+	case exclude:
+		for k := range result {
+			for e := range keySet {
+				if strings.Contains(k, strings.ToLower(e)) {
+					delete(result, k)
+				}
+			}
+		}
```

Comment:
I thought we going for exact match? We are still doing `strings.Contains` here

Reply:
@agnivade it was never exact match for exclusion case, do you think we should also change that?